### PR TITLE
MGMT-14375: Setup appliance local registry

### PR DESCRIPTION
**Pull request by nmagnezi**
@danielerez

### CODE REVIEW

File: pkg/asset/data/data_iso.go [hunk 1]
```diff
@@ -110,3 +113,18 @@ func (a *DataISO) updateAsset(envConfig *config.EnvConfig) error {
 
 	return nil
 }
+
+func (a *DataISO) copyRegistryImageIfNeeded(envConfig *config.EnvConfig) error {
+	imagesDirPath := filepath.Join(envConfig.TempDir, dataDir, imagesDir)
+
+	// Search for Image in temp dir
+	if fileName := envConfig.FindInTemp(filepath.Join(dataDir, imagesDir, templates.RegistryFilePath)); fileName != "" {
+		logrus.Info("Reusing registry.tar from temp")
```
Comment (danielerez) [inline pkg/asset/data/data_iso.go:122]: need to rephrase a bit as it's exposed to the user, or just log.debug for now

File: pkg/registry/registry.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func (r *registry) StartRegistry(registryDataPath string) error {
+	_ = r.StopRegistry()
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	filePath := filepath.Join(pwd, registryDataPath, bootstrapRegistryDir)
```
Comment (danielerez) [inline pkg/registry/registry.go:43]: this should be generic (bootstrapRegistryDir passed to api...)

File: pkg/registry/registry.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+func NewRegistry() Registry {
+	return &registry{
+		executer: executer.NewExecuter(),
+	}
+}
+
+func (r *registry) StartRegistry(registryDataPath string) error {
+	_ = r.StopRegistry()
+	pwd, err := os.Getwd()
```
Comment (danielerez) [inline pkg/registry/registry.go:38]: we can try later filepath.Abs instead (maybe a bit cleaner)

File: pkg/registry/registry.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	"strings"
+
+	"github.com/danielerez/openshift-appliance/pkg/executer"
+	"github.com/danielerez/openshift-appliance/pkg/templates"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	registryStartCmd     = "podman run --privileged -d --name registry -p 5000:5000 -p 443:5000 -v %s:/var/lib/registry --restart=always -e REGISTRY_HTTP_ADDR=0.0.0.0:5000 %s"
```
Comment (danielerez) [inline pkg/registry/registry.go:16]: no need for "-p 443:5000". maybe also no need for REGISTRY_HTTP_ADDR=0.0.0.0:5000 (?)

File: pkg/registry/registry.go (hunk 3 above)
Comment (nmagnezi) [reply pkg/registry/registry.go:37]: I see `filepath.Abs` also returns an error:
go
```
func Abs(path string) (string, error) {
	return abs(path)
}
```

File: pkg/registry/registry.go (hunk 4 above)
Comment (nmagnezi) [reply pkg/registry/registry.go]: Done

File: pkg/registry/registry.go (hunk 2 above)
Comment (nmagnezi) [reply pkg/registry/registry.go]: Done

File: pkg/asset/data/data_iso.go (hunk 1 above)
Comment (nmagnezi) [reply pkg/asset/data/data_iso.go]: Done

**Pull request merged**